projects/lexical_shared/lib/lexical/document/store.ex

```diff
@@ -258,7 +259,15 @@ defmodule Lexical.Document.Store do
 
   @spec open(Lexical.uri(), String.t(), pos_integer()) :: :ok | {:error, :already_open}
   def open(uri, text, version) do
-    GenServer.call(name(), {:open, uri, text, version})
+    language_id = Document.Path.language_id_from_uri(uri)
```

Comment:
can we push this down into the document store by defaulting to nil here?

Reply:
Yep, done.